server/internal/usecase/interactor/request.go

```diff
@@ -0,0 +1,213 @@
+			Workspace(ws.ID()).
+			Project(param.ProjectID).
+			CreatedBy(*operator.User).
+			Thread(th.ID()).
+			Items(param.Items).
+			Title(param.Title)
+
+		if param.State != nil {
+			if *param.State == request.StateApproved || *param.State == request.StateClosed {
+				return nil, fmt.Errorf("can't create request with state %v", param.State.String())
+			}
+			builder.State(*param.State)
+		}
+		if param.Description != nil {
+			builder.Description(*param.Description)
+		}
+		if param.Reviewers != nil && param.Reviewers.Len() > 0 {
+			for _, rev := range param.Reviewers {
+				if !ws.Members().IsOwnerOrMaintainer(rev) {
+					return nil, errors.New("reviewer should be owner or maintainer")
+				}
+			}
+			builder.Reviewers(param.Reviewers)
+		}
+
+		req, err := builder.Build()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := r.repos.Request.Save(ctx, req); err != nil {
+			return nil, err
+		}
+
+		return req, nil
+	})
+}
+
+func (r Request) Update(ctx context.Context, param interfaces.UpdateRequestParam, operator *usecase.Operator) (*request.Request, error) {
+	if operator.User == nil {
+		return nil, interfaces.ErrInvalidOperator
+	}
+
+	return Run1(ctx, operator, r.repos, Usecase().Transaction(), func() (*request.Request, error) {
+		req, err := r.repos.Request.FindByID(ctx, param.RequestID)
+		if err != nil {
+			return nil, err
+		}
+		ws, err := r.repos.Workspace.FindByID(ctx, req.Workspace())
+		if err != nil {
+			return nil, err
+		}
+		// only owners, maintainers, and the request creator can update requests
+		canUpdate := *operator.User == req.CreatedBy() || ws.Members().IsOwnerOrMaintainer(*operator.User)
+		if !operator.IsWritableWorkspace(req.Workspace()) && canUpdate {
+			return nil, interfaces.ErrOperationDenied
+		}
+
+		if param.State != nil {
+			if *param.State == request.StateApproved {
+				return nil, errors.New("can't update by approve")
+			}
+			req.SetState(*param.State)
+		}
+		if param.Description != nil {
+			req.SetDescription(*param.Description)
+		}
+		if param.Reviewers != nil && param.Reviewers.Len() > 0 {
+			for _, rev := range param.Reviewers {
+				if !ws.Members().IsOwnerOrMaintainer(rev) {
+					return nil, errors.New("reviewer should be owner or maintainer")
+				}
+			}
+			req.SetReviewers(param.Reviewers)
+		}
+		if param.Items != nil {
+			req.SetItems(param.Items)
+		}
+
+		if err := r.repos.Request.Save(ctx, req); err != nil {
+			return nil, err
+		}
+
+		return req, nil
+	})
+}
+
+func (r Request) Delete(ctx context.Context, requestID id.RequestID, operator *usecase.Operator) error {
+	if operator.User == nil {
+		return interfaces.ErrInvalidOperator
+	}
+
+	return Run0(ctx, operator, r.repos, Usecase().Transaction(), func() error {
+		req, err := r.repos.Request.FindByID(ctx, requestID)
+		if err != nil {
+			return err
+		}
+
+		if !operator.IsWritableProject(req.Project()) {
+			return interfaces.ErrOperationDenied
+		}
+
+		return r.repos.Request.Remove(ctx, requestID)
+	})
```

Comment:
```suggestion
	return r.repos.Request.Remove(ctx, requestID)
```

Already filter is applied